File: bitlap-core/src/main/kotlin/org/bitlap/core/sql/BitlapSqlPlan.kt
/**
 * Copyright (C) 2023 bitlap.org .
 */
package org.bitlap.core.sql

import org.apache.calcite.rel.RelNode
import org.apache.calcite.sql.SqlNode

/**
 * sql plan
 */
data class BitlapSqlPlan(
    val statement: String,
    val sqlNode: SqlNode,
    val rel: RelNode,
    val relOpt: RelNode, // optimized rel
) {

    fun explain(): String {
        return this.relOpt.explain()
    }
}

File: bitlap-core/src/main/scala/org/bitlap/core/sql/BitlapSqlPlan.scala
/**
 * Copyright (C) 2023 bitlap.org .
 */
package org.bitlap.core.sql

import org.apache.calcite.rel.RelNode
import org.apache.calcite.sql.SqlNode

/** sql plan
 */
final case class BitlapSqlPlan(
  statement: String,
  sqlNode: SqlNode,
  rel: RelNode,
  relOpt: RelNode // optimized rel
) {

  def explain(): String = {
    this.relOpt.explain()
  }
}
